File: ts/Person.ts
/**
 * 人間クラス
 */
export class Person{

	/** 名前 */
	firstName: string;
	/** 苗字 */
	lastName: string;
	/** 年齢 */
	age: number;
	/** 身長 */
	height: number;

	/**
	 * 人間クラスのコンストラクタ
	 * @remarks
	 * 人間クラスを初期化します。
	 * @param firstName - 名前
	 * @param lastName - 苗字
	 * @param age - 年齢
	 * @param height - 身長
	 * @returns 人間クラスのオブジェクト
	 */
	constructor(firstName: string, lastName: string, age: number, height: number){ 
		this.firstName = firstName;
		this.lastName = lastName;
		this.age = age;
		this.height = height;
	}

	/**
	 * フルネームを取得
	 * @remarks
	 * フルネームを取得します。
	 * @param isReverse - 取得するフルネームを逆順にする
	 * @param delimiter - 区切り文字
	 * @returns フルネームの文字列
	 */
	fullName(isReverse: boolean = false, delimiter: string = " "){
		return isReverse?
			this.lastName + delimiter + this.firstName:
			this.firstName + delimiter + this.lastName;
	}

	/**
	 * ステータスを表示
	 * @remarks
	 * ステータスを表示します。
	 */
	 viewStatus(){
	 	console.log(`---- STATUS ----
FIRST NAME: ${this.firstName}
LAST NAME : ${this.lastName}
AGE       : ${this.age} 
HEIGHT    : ${this.height} cm`
		);
	}

}

File: hsp/doclib/Person/Person.hs
%dll
Person

%ver
3.6

%date
2023/7/17

%author
YOS G-spec

%url
https://github.com/yosgspec

%note
人間モジュール

%type
ユーザー定義命令/関数

%port
Win
Mac
Cli
HSP3Dish

%group
人間モジュール

%index
new@Person
人間モジュールのコンストラクタ

%prm
(psn, firstName, lastName, age, height)
psn : Personモジュールで初期化する変数
firstName : 名前(文字列)
lastName : 苗字(文字列)
age : 年齢(整数)
height : 身長(浮動小数点)

%inst
人間モジュールを初期化します。

%href
firstName
lastName
age
height
fullName
viewStatus
%index
fullName
フルネームを取得

%prm
(isReverse, delimita)
val : フルネーム(文字列)
isReverse=0 : 取得するフルネームを逆順にする(1=有効化, 2=無効化)
delimiter=" " : 区切り文字列

%inst
フルネームを取得します。

%href
new@Person
firstName
lastName
%index
firstName
名前を取得

%prm
(psn)
val : 名前(文字列)
psn : Personのモジュール型変数

%inst
名前を取得します。

%href
new@Person
%index
lastName
苗字を取得

%prm
(psn)
val : 苗字(文字列)
psn : Personのモジュール型変数

%inst
苗字を取得します。

%href
new@Person
%index
age
年齢を取得

%prm
(psn)
val : 年齢(整数)
psn : Personのモジュール型変数

%inst
年齢を取得します。

%href
new@Person
%index
viewStatus
ステータスを表示

%inst
ステータスを表示します。

%href
new@Person
firstName
lastName
age
height

%index
height
身長を取得

%prm
(psn)
val : 身長(浮動小数点)
psn : Personのモジュール型変数

%inst
身長を取得します。

%href
new@Person
